src/source/PeerConnection/PeerConnection.c

```diff
@@ -1065,12 +1069,19 @@ STATUS freePeerConnection(PRtcPeerConnection* ppPeerConnection)
     }
 
     if (pKvsPeerConnection->pTwccManager != NULL) {
+        MUTEX_LOCK(pKvsPeerConnection->twccLock);
+        twccLocked = TRUE;
+        hashTableGetCount(pKvsPeerConnection->pTwccManager->pTwccRtpPktInfosHashTable, &twccHashTableCount);
+        DLOGI("Number of TWCC info packets in memory: %d", twccHashTableCount);
+        hashTableIterateEntries(pKvsPeerConnection->pTwccManager->pTwccRtpPktInfosHashTable, 0, freeHashEntry);
```

Comment:
`hashTableGetCount` and `hashTableIterateEntries` return a `STATUS` so we should check for success.

Once you add that, then you will need to make sure to add an `if locked` section in the CleanUp as @jdelapla suggested.

Reply:
Done. Already did the if(locked) check before freeing the twcc mutex.